File: Data Structures & Algorithms/min-cost-climbing-stairs/submission-0.py
class Solution:
    def minCostClimbingStairs(self, cost: List[int]) -> int:
        def solve(x) : 
            if  x == 0 or x == 1 :
                return 0

            return min(solve(x - 1) + cost[x - 1], solve(x - 2) + cost[x - 2])
        
        return solve(len(cost))

File: Data Structures & Algorithms/max-water-container/submission-0.cpp
class Solution {
public:
    int maxArea(vector<int>& heights) {
        int left = 0, right = heights.size() - 1;
        int result = 0;
        while (left < right) {
            result = max(result, min(heights[left], heights[right]) * (right - left));

            if (heights[left + 1] > heights[right - 1]) left++;
            else if (right > left) right--;
        }

        return result ;
    }
};
